agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java

```diff
@@ -118,16 +119,19 @@ private void export(SpanData span) {
             boolean isLog = removeAttributeBoolean(attributes, AI_INTERNAL_LOG);
             if (isLog) {
                 exportLogSpan(span, attributes);
-            } else if (!SpanId.isValid(span.getParentSpanId())) {
-                // TODO revisit this decision
-                // maybe user-generated telemetry?
-                // otherwise this top-level span won't show up in Performance blade
+            } else if ("spring-scheduling".equals(stdComponent) && !SpanId.isValid(span.getParentSpanId())) {
+                // TODO need semantic convention for determining whether to map INTERNAL to request or dependency
+                //  (or need clarification to use SERVER for this)
                 exportRequest(span, attributes);
             } else {
                 exportRemoteDependency(span, attributes, true);
             }
         } else if (kind == Kind.CLIENT || kind == Kind.PRODUCER) {
             exportRemoteDependency(span, attributes, false);
+        } else if (kind == Kind.CONSUMER && !span.getHasRemoteParent()) {
```

Comment:
getHasRemoteParent, can we fix it in the opentelemetry-api? it doesn't seem to follow the naming convention for boolean getter.

Reply:
good point! I just opened https://github.com/open-telemetry/opentelemetry-java/issues/1910